config/application.rb

```diff
@@ -104,7 +104,7 @@ class Application < Rails::Application
     Dir["#{Rails.root}/config/routes/**/*.rb"].each do |route_file|
       config.paths['config/routes.rb'] << route_file
     end
-
+    config.load_defaults 6.0
```

Comment:
I had to add
```ruby
    # Rails 5 changed this to true
    config.active_record.belongs_to_required_by_default = false
```

Reply:
I opened https://github.com/theforeman/foreman/pull/9382 to do this as a separate preparation step.